assets/src/infrastructure/ui/inputs/style.scss

```diff
@@ -0,0 +1,79 @@
+.input-select-textarea-hover {
+	border-color: var(--ee-color-primary-low-contrast);
+	color: var(--ee-color-primary-high-contrast);
+	box-shadow: 0 0 0 1px var(--ee-color-primary-low-contrast);
+}
+
+.input-select-textarea-focus {
+	border-color: var(--ee-color-primary);
+	box-shadow: 0 0 0 1px var(--ee-color-primary);
+}
+
+.ee-input-base {
+	&.ee-input-base {
+		border-color: var(--ee-border-color);
+		border-style: solid;
+		border-width: 2px;
+		&:focus {
+			/* Visible in Windows high-contrast themes */
+			outline-color: transparent;
+			outline-width: 2px;
+			outline-style: solid;
+		}
+	}
+
+	&.ee-date-picker.ee-date-picker {
+		input[type='text'] {
+			&:hover {
+				@extend .input-select-textarea-hover;
+			}
+			&:focus {
+				@extend .input-select-textarea-focus;
+			}
+		}
+	}
+
+	&.ee-input {
+		@at-root {
+			input[type='text']#{&},
```

Comment:
I tried to avoid that `@at-root`, but as I wrote this is kind of last resort.

The ampersand works with class names, not the same with HTML tags:

![image](https://user-images.githubusercontent.com/1477453/80588117-64072d80-8a20-11ea-96db-0bb89f9f368e.png)

Reply:
another solution might be to wrap input with a div and then have this:
```
	&.ee-input {
		.input-div-wrapper {
			input[type='text'] {
```
but I didn't went with this approach for 2 reasons:
1. tried to avoid additional markup
***
2. for example chakra's `NumberInput` composition is following:
```
<NumberInput>
  <NumberInputField />
  <NumberInputStepper>
    <NumberIncrementStepper />
    <NumberDecrementStepper />
  </NumberInputStepper>
</NumberInput>
```
to do that wrapping we'd have:
```
<NumberInput>
  <div className="number-input-wrapper">
    <NumberInputField />
  </div>
  <NumberInputStepper>
    <NumberIncrementStepper />
    <NumberDecrementStepper />
  </NumberInputStepper>
</NumberInput>
```
but this might break composition pattern, since internally chrakra passes down some props to direct child, didn't testes with this case, but I notice this in other case
***
I am open for an alternative to `@at-root`

Reply:
another solution might be to wrap input with a div and then have this:
```
	&.ee-input {
		.input-div-wrapper {
			input[type='text'] {
```
but I didn't went with this approach for 2 reasons:
1. tried to avoid additional markup
***
2. for example chakra's `NumberInput` composition is following:
```
<NumberInput>
  <NumberInputField />
  <NumberInputStepper>
    <NumberIncrementStepper />
    <NumberDecrementStepper />
  </NumberInputStepper>
</NumberInput>
```
to do that wrapping we'd have:
```
<NumberInput>
  <div className="number-input-wrapper">
    <NumberInputField />
  </div>
  <NumberInputStepper>
    <NumberIncrementStepper />
    <NumberDecrementStepper />
  </NumberInputStepper>
</NumberInput>
```
but this might break composition pattern, since internally chrakra passes down some props to direct child, didn't testes with this case, but I notice this in other case
***
I am open for an alternative to `@at-root`

Reply:
oops, just realized the screenshot from sass playground was wrong, this is with that suggested implementation:

![image](https://user-images.githubusercontent.com/1477453/80590566-c95d1d80-8a24-11ea-8125-464b5d2acd5b.png)
***
having one space between selector and ampersand will make it parent:

![image](https://user-images.githubusercontent.com/1477453/80590653-f3aedb00-8a24-11ea-946d-a4e035c5ab47.png)